server/issue.go

```diff
@@ -521,12 +521,18 @@ func (p *Plugin) httpGetJiraProjectMetadata(w http.ResponseWriter, r *http.Reque
 	projects := []utils.ReactSelectOption{}
 	issues := map[string][]utils.ReactSelectOption{}
 	for _, prj := range plist {
+		issueTypeList, err := p.GetIssueTypes(types.ID(instanceID), types.ID(mattermostUserID), prj.ID)
+		if err != nil {
+			p.API.LogError("Failed to get issue types for project.", "ProjectKey", prj.Key, "Error", err.Error())
```

Comment:
@Kshitij-Katiyar This list of projects is potentially very large. With the current implementation here, we will run into an `n+1` problem.

What do you think about my suggestion here? https://github.com/mattermost/mattermost-plugin-jira/issues/912#issuecomment-1430341080

> If we run into the error (does it tell us which projects?), then fetch the projects without expanding the issue type. Then individually fetch issue types for each project. This is not performant, but may be necessary if we don't know which projects have issue types.

I'm thinking we should first attempt to fetch all projects with one request like we used to be doing, and if that throws an error then fetch them without the expand, and then loop through them and fetch issue types like this, but only when that error occurs.

Another question: Are we able to fetch the projects without issue types, and determine if they have issue types somehow, without triggering the Jira error?

Reply:
Please take into account how this affects Jira Cloud. We don't want to decrease performance there while fixing a bug for Jira Server, so we should by default still be fetching everything with one request for Jira Cloud.

Reply:
@mickmister No it does not tell us which project is triggering the error if we fetch the project list with issueTypes expanded.
Yes, we can first fetch the project with issueTypes expanded and if we run into an error then we can fetch issueTypes individually for each project. Just have one question here should we fetch in all errors or just when we will get this particular error?
And we cannot determine in the project list whether they have issueTypes or not without calling any additional API.

Reply:
> Just have one question here should we fetch in all errors or just when we will get this particular error?

@Kshitij-Katiyar I trust your judgment on this. I'm leaning towards "just that one kind of error". Note that this error could be in another language than English, so hopefully there is an error id or something like that available.

Reply:
@mickmister I am not able to find any error id associated with this but we can do this for this particular status code. 
One more question should I remove the changes for the cloud part?

Reply:
@Kshitij-Katiyar It's probably best to take the defensive approach and include it for Cloud as well. One thing I want to emphasize is we should avoid the n+1 situation in any way possible. I realize it's required in some scenarios though.

Reply:
@Kshitij-Katiyar I think we should continue fetching the issue types with `expand` for both Server and Cloud, and fallback on `GetIssueTypes` only when we need to

Reply:
sure @mickmister